### PR TITLE
CATTY-241 Add Swift extension for ProjectDetailStoreViewController

### PR DESCRIPTION
**Pull request by DhrubojyotiBis1**
- Created a new Swift extension for ProjectDetailStoreViewController
- Moved ProjectDetailStoreViewController to the same folder (Details/)
- Converted the code from CreateView to the newly created Swift extension
- Removed the old CreateView

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+            label?.font = UIFont.boldSystemFont(ofSize: 24)
+        } else {
+            label?.font = UIFont.boldSystemFont(ofSize: 17)
+        }
+        label?.textColor = UIColor.globalTint
+        label?.layer.shadowColor = UIColor.white.cgColor
+        label?.layer.shadowOffset = CGSize(width: 0.0, height: 0.0)
+    }
+
+    func configureAuthorLabel(_ label: UILabel?, andHeight height: CGFloat) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:98]: Please remove this function and use `configureTextLabel` instead.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+            label?.font = UIFont.boldSystemFont(ofSize: 18)
+        } else {
+            label?.font = UIFont.boldSystemFont(ofSize: 12)
+        }
+        label?.textColor = UIColor.textTint
+        label?.layer.shadowColor = UIColor.white.cgColor
+        label?.layer.shadowOffset = CGSize(width: 0.0, height: 0.0)
+    }
+
+    func configureTextLabel(_ label: UILabel?, andHeight height: CGFloat) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:110]: Please remove optionals for both parameters and set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+            label?.font = UIFont.boldSystemFont(ofSize: 18)
+        } else {
+            label?.font = UIFont.boldSystemFont(ofSize: 12)
+        }
+        label?.textColor = UIColor.textTint
+        label?.layer.shadowColor = UIColor.white.cgColor
+        label?.layer.shadowOffset = CGSize(width: 0.0, height: 0.0)
+    }
+
+    func setMaxHeightIfGreaterFor(_ view: UIView?, withHeight height: CGFloat) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:122]: The `UIView` should not be optional and set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 4]
```diff
@@ -0,0 +1,411 @@
...
+        var frame = view?.frame
+        if (frame?.size.height ?? 0.0) < height {
+            frame?.size.height = height
+            view?.frame = frame ?? CGRect.zero
+        }
+    }
+
+    func configureDescriptionLabel(_ label: TTTAttributedLabel?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:130]: The parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 5]
```diff
@@ -0,0 +1,411 @@
...
+        var mutableActiveLinkAttributes: [AnyHashable: Any] = [:]
+        mutableActiveLinkAttributes[kCTForegroundColorAttributeName as String] = UIColor.brown
+        mutableActiveLinkAttributes[kCTUnderlineStyleAttributeName as String] = NSNumber(value: false)
+        if let label = label {
+            label.linkAttributes = mutableLinkAttributes
+            label.activeLinkAttributes = mutableActiveLinkAttributes
+        }
+    }
+
+    func addShadowToTitleLabel(for button: UIButton?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:151]: I think this method is unused.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 6]
```diff
@@ -0,0 +1,411 @@
...
+        button?.titleLabel?.layer.shadowColor = UIColor.black.cgColor
+        button?.titleLabel?.layer.shadowOpacity = 0.3
+        button?.titleLabel?.layer.shadowRadius = 1
+        button?.titleLabel?.layer.shadowOffset = CGSize(width: 0.0, height: 2.0)
+    }
+
+    func addHorizontalLine(to view: UIView?, andHeight height: CGFloat) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:158]: The parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 7]
```diff
@@ -0,0 +1,411 @@
...
+        self.setMaxHeightIfGreaterFor(view, withHeight: height)
+        let offset = (view?.frame.size.height ?? 0.0) + 1
+        let lineView = UIView(frame: CGRect(x: (view?.frame.size.width ?? 0.0) / 15 - 10, y: offset, width: view?.frame.size.width ?? 0.0, height: 1))
+        lineView.backgroundColor = UIColor.utilityTint
+        view?.addSubview(lineView)
+    }
+
+    func getInformationTitleLabel(withTitle icon: UIImage?, atXPosition xPosition: CGFloat, atYPosition yPosition: CGFloat, andHeight height: CGFloat) -> UIImageView? {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:166]: The parameter and return value should not be optional. Also, set the scope to fileprivate.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 8]
```diff
@@ -0,0 +1,411 @@
...
+
+        let titleInformation = UIImageView(frame: CGRect(x: xPosition, y: yPosition, width: 15, height: 15))
+        titleInformation.image = icon
+
+        return titleInformation
+    }
+
+    func getInformationDetailLabel( withTitle title: String?, atXPosition xPosition: CGFloat, atYPosition yPosition: CGFloat, andHeight height: CGFloat) -> UILabel? {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:174]: The parameter and return value should not be optional. Also, set the scope to fileprivate.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 9]
```diff
@@ -0,0 +1,411 @@
...
+        } else {
+            detailInformationLabel.font = UIFont.systemFont(ofSize: 14.0)
+        }
+
+        detailInformationLabel.backgroundColor = UIColor.clear
+        detailInformationLabel.sizeToFit()
+        return detailInformationLabel
+    }
+
+    func addReportButton(to view: UIView?, withTarget target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:193]: Please remove optionals for both parameters and set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 10]
```diff
@@ -0,0 +1,411 @@
...
+            reportButton.tintColor = UIColor.buttonTint
+            reportButton.setTitleColor(UIColor.buttonTint, for: .normal)
+            if let view = view {
+                view.addSubview(reportButton)
+                setMaxHeightIfGreaterFor(view, withHeight: view.frame.size.height + reportButton.frame.size.height)
+            }
+        }
+    }
+
+    func addInformationLabel(to view: UIView?, withAuthor author: String?, downloads: NSNumber?, uploaded: String?, version: String?, views: NSNumber?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:213]: Please remove optionals for both parameters and set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 11]
```diff
@@ -0,0 +1,411 @@
...
+                view?.addSubview(infoLabel)
+            }
+
+            offset += +height * 0.04
+            counter += 1
+        }
+        setMaxHeightIfGreaterFor(view, withHeight: offset)
+    }
+
+    func addLoadingButton(to view: UIView?, withTarget target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:255]: The `view` parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 12]
```diff
@@ -0,0 +1,411 @@
...
+        button.tintColor = UIColor.buttonTint
+        let xPosition = (view?.frame.size.width ?? 0.0) - 40
+        let yPosition = (view?.frame.size.height ?? 0.0) * 0.1 + Util.screenHeight() / 4.5
+        button.frame = CGRect(x: xPosition, y: yPosition - 25, width: 28, height: 28)
+        button.isHidden = true
+        button.addTarget(target, action: #selector(URLProtocol.stopLoading), for: .touchUpInside)
+        view?.addSubview(button)
+    }
+
+    func addDownloadAgainButton(to view: UIView?, withTarget target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:267]: The `view` parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 13]
```diff
@@ -0,0 +1,411 @@
...
+            downloadAgainButton.titleLabel?.adjustsFontSizeToFitWidth = true
+            downloadAgainButton.titleLabel?.minimumScaleFactor = 0.4
+            downloadAgainButton.addTarget(target, action: #selector(downloadAgain), for: .touchUpInside)
+            downloadAgainButton.tag = Int(kDownloadAgainButtonTag)
+            downloadAgainButton.isHidden = true
+            view?.addSubview(downloadAgainButton)
+        }
+    }
+
+    func addOpenButton(to view: UIView?, withTarget target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:286]: The `view` parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 14]
```diff
@@ -0,0 +1,411 @@
...
+            openButton.titleLabel?.adjustsFontSizeToFitWidth = true
+            openButton.titleLabel?.minimumScaleFactor = 0.4
+            openButton.setTitle(kLocalizedOpen, for: .normal)
+            openButton.addTarget(target, action: #selector(openButtonPressed(_:)), for: .touchUpInside)
+            openButton.tintColor = UIColor.buttonTint
+            view?.addSubview(openButton)
+        }
+    }
+
+    func addDownloadButton(to view: UIView?, withTarget target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:304]: The `view` parameter should not be optional. Also, set the scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 15]
```diff
@@ -0,0 +1,411 @@
...
+            let activity = UIActivityIndicatorView(style: .gray)
+            activity.tag = Int(kActivityIndicator)
+            activity.frame = CGRect(x: 5, y: 0, width: 25, height: 25)
+            downloadButton.addSubview(activity)
+
+            view?.addSubview(downloadButton)
+        }
+    }
+
+    func addThumbnailImage(withImageUrlString imageUrlString: String?, to view: UIView?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:328]: Both parameters should not be optional. Also set scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift [hunk 16]
```diff
@@ -0,0 +1,411 @@
...
+        imageView.contentMode = .scaleAspectFill
+        imageView.layer.cornerRadius = 8.0
+        imageView.layer.masksToBounds = true
+        imageView.layer.borderColor = UIColor.utilityTint.cgColor
+        imageView.layer.borderWidth = 1.0
+
+        view?.addSubview(imageView)
+    }
+
+    func addProjectDescriptionLabel(withDescription description: String?, to view: UIView?, target: Any?) {
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewControllerExtension.swift:361]: `description` and `view` should not be optional. Also set scope to `fileprivate`.

File: src/Catty/ViewController/Download/Details/ProjectDetailStoreViewController.h [hunk 17]
```diff
@@ -32,6 +32,8 @@
 
 @property (nonatomic, strong) CatrobatProject *project;
 @property (nonatomic, strong) NSMutableDictionary *projects;
+- (void)reportProject;
+-(void)downloadAgain;
```
Comment (m-herold) [inline src/Catty/ViewController/Download/Details/ProjectDetailStoreViewController.h:36]: Please add space after `-` to have a consistent style.

**Comment by DhrubojyotiBis1** [comment]
Hi, @m-herold  I have made the changes. Please review

**Pull request closed without merging**